gulpfile.js

```diff
@@ -88,34 +88,35 @@ var paths = {
   },
 
   vendorChallenges: [
-    'public/bower_components/jshint/dist/jshint.js',
-    'public/bower_components/chai/chai.js',
-    'public/bower_components/CodeMirror/lib/codemirror.js',
-    'public/bower_components/CodeMirror/addon/comment/comment.js',
-    'public/bower_components/CodeMirror/addon/edit/closebrackets.js',
-    'public/bower_components/CodeMirror/addon/edit/matchbrackets.js',
-    'public/bower_components/CodeMirror/addon/lint/lint.js',
-    'public/bower_components/CodeMirror/addon/lint/javascript-lint.js',
-    'public/bower_components/CodeMirror/mode/javascript/javascript.js',
-    'public/bower_components/CodeMirror/mode/xml/xml.js',
-    'public/bower_components/CodeMirror/mode/css/css.js',
-    'public/bower_components/CodeMirror/mode/htmlmixed/htmlmixed.js',
+    'node_modules/jshint/dist/jshint.js',
+    'node_modules/chai/chai.js',
+    'node_modules/codemirror/lib/codemirror.js',
+    'node_modules/codemirror/addon/comment/comment.js',
+    'node_modules/codemirror/addon/edit/closebrackets.js',
+    'node_modules/codemirror/addon/edit/matchbrackets.js',
+    'node_modules/codemirror/addon/lint/lint.js',
+    'node_modules/codemirror/addon/lint/javascript-lint.js',
+    'node_modules/codemirror/mode/javascript/javascript.js',
+    'node_modules/codemirror/mode/xml/xml.js',
+    'node_modules/codemirror/mode/css/css.js',
+    'node_modules/codemirror/mode/htmlmixed/htmlmixed.js',
```

Comment:
> require.resolve() can be effective when calling from code, but when called from top scope, it just returns the global node_modules path for the module if it finds one.

I'm not sure what you mean by that statement.

> require.resolve('react')
'/Users/berkeley/node/rwr/actual/node_modules/react/react.js'
> require.resolve('react-router')
'/Users/berkeley/node/rwr/actual/node_modules/react-router/lib/index.js'
> require.resolve('warning')
'/Users/berkeley/node/rwr/actual/node_modules/warning/warning.js'

Here I use the `require.resolve` method to deliver the directory of the main file for a module according to the rules set by node. 

This can be used to find the directory relative to `node_modules`. With a little string concatenation magic we have the correct path to the correct file.